### ISSUE TITLE
Disentangle the dosdp pattern phony-target hairball

### ISSUE THREAD

**Issue by matentzn** [INCATools/ontology-development-kit#448]
The DOSDP pipeline needs to be disentangled - too many phony goals basically force rebuilding of downstream artefacts.

**Comment by matentzn** [comment]
We currently have this situation:

```
definitions.owl -> data/default/mypattern.owl -> imports/go_import.owl -> [ mirror/go_mirror.owl | data/default/mypattern.owl]
```

which means there is this terrible cycle:

```
 data/default/mypattern.owl -> imports/go_import.owl ->  data/default/mypattern.owl
```

This is because dosdp-tools needs the GO terms in the module to generate (and later to match), but the module cannot be extracted without knowing what terms to import.

There are various hacks you can do to improve this, like hacking into `data/default/mypattern.tsv` to get to the required seed, but I no longer think this is a good idea. @ramonawalls has repeatedly pointed out the annoyance that rebuilding the patterns re-triggers also the imports. 

I now believe that there should _not be a make dependency between  `data/default/mypattern.owl` and `imports/go_import.owl`. In much the same way as when creating ROBOT template, we can train people to understand this: 

You run the pipeline once, yes, you will have a few unresolved external references in your `definitions.owl`. You run it twice, all good (the signature of definitions.owl is now added to the seed to extract the module). The cost of running twice is _much lower_ then having this magic step in there and waiting for the import to be recreated _regardless of whether it is necessary or not_.

@dosumis @ramonawalls @cmungall I would like your take on this. Again, their are hacks to get the patterns to build right in one go, but I think training will achieve a better effect in the end.

**Comment by dosumis** [comment]
I think this is the right approach.  Presumably you could also fix by running imports separately later.  The one possible downside is missing inconsistencies coming from imports on PR CI test.

**Comment by balhoff** [comment]
@matentzn do you use `dosdp-tools terms`? That command is supposed to allow you to get the signature without running `generate`.

**Comment by matentzn** [comment]
@balhoff forgot about that. But the problem is the same: you have to refresh all imports hoping to hit the one you need to suck in the new terms, and even if you could guess the import based on the prefixes (hack hack hack) you still have to run the import every time - wether you need to or not.

**Comment by dosumis** [comment]
> even if you could guess the import based on the prefixes (hack hack hack) you still have to run the import every time - wether you need to or not.

Not necessarily.  Should be possible to list terms from `dosdp-tools terms` that are not in the signature of the ontology and only update imports for the ontologies they come from.

**Comment by matentzn** [comment]
Yes, but you will still have to guess, i.e. hope, that GO:123 is in go_import.owl. In there is nothing preventing the user from renaming an import, e.g. you can define an import `go_plus` or `biological_process`, which will make this guesswork from ontology term to ontology even more guessy. Building such prefix assumptions into the pipeline is hacky whatever you do.. think GOREL, NCBITAXONUNION, and dozens of other examples.

**Comment by matentzn** [comment]
Fixed by #530

**Comment by ramonawalls** [comment]
Excellent! Thank you.
